File: csharp/2016/the-girl-with-all-the-gifts-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "The Girl with All the Gifts" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "The Girl with All the Gifts",
                year: 2016,
                plot: "A scientist and a teacher living in a dystopian future embark on a journey of survival with a special young girl named Melanie.",
                rating: 6.7
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "The Girl with All the Gifts",
                year: 2016
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}

File: python/2016/the-girl-with-all-the-gifts-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add The Girl with All the Gifts to the database
movies.insert(
    title="The Girl with All the Gifts", 
    year=2016, 
    plot="A scientist and a teacher living in a dystopian future embark on a journey of survival with a special young girl named Melanie.", 
    rating=6.7
)

# Confirm that the movie was added
movie = movies.select(
    title="The Girl with All the Gifts", 
    year=2016
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    